jcache/commons/src/main/java/org/infinispan/jcache/AbstractJCacheNotifier.java

```diff
@@ -37,6 +42,13 @@
 
    private static final boolean isTrace = log.isTraceEnabled();
 
+   private static final ConcurrentLinkedQueue<CountDownLatch> EMPTY_QUEUE = new ConcurrentLinkedQueue<CountDownLatch>() {
```

Comment:
Right, but neither the type definition nor the implementation needs to be `ConcurrentLinkedQueue` for that. ConcurrentLinkedQueue is an implementation. The type might be `Queue` or `Collection`, preferring the latter. If you need the former, you can simply implement an empty queue yourself, without instantiating ConcurrentLinkedQueue. If you need the latter, there's helper utilities already.